engine/common/grpc/compressor/lz4/lz4.go

```diff
@@ -0,0 +1,68 @@
+package lz4
+
+import (
+	"io"
+	"sync"
+
+	"github.com/pierrec/lz4"
+	"google.golang.org/grpc/encoding"
+)
+
+// Name is the name registered for the LZ4 compressor.
+const Name = "lz4"
```

Comment:
given the high memory utilization of lz4, let's not include it as an available option. I worry it could be used as an resource attack vector.